src/Microsoft.AspNet.Identity/IIdentityKeyNormalizer.cs

```diff
@@ -4,15 +4,15 @@
 namespace Microsoft.AspNet.Identity
 {
     /// <summary>
-    ///     Used to normalize a user name
+    ///     Used to normalize keys for consistent lookups
     /// </summary>
-    public interface IUserNameNormalizer
+    public interface IIdentityKeyNormalizer
```

Comment:
How about we just drop string and key then, and just do IIdentityLookupNormalizer?